tests/base.py

```diff
@@ -245,46 +249,56 @@ def max_bookmarks_by_stream(self, sync_records):
         """
         max_bookmarks = {}
         for stream, batch in sync_records.items():
-
-            upsert_messages = [m for m in batch.get('messages') if m['action'] == 'upsert']
-            stream_bookmark_key = self.expected_replication_keys().get(stream, set())
-            assert len(stream_bookmark_key) == 1  # There shouldn't be a compound replication key
-            stream_bookmark_key = stream_bookmark_key.pop()
-
-            bk_values = [message["data"].get(stream_bookmark_key) for message in upsert_messages]
-            max_bookmarks[stream] = {stream_bookmark_key: None}
-            for bk_value in bk_values:
-                if bk_value is None:
-                    continue
-
-                if max_bookmarks[stream][stream_bookmark_key] is None:
-                    max_bookmarks[stream][stream_bookmark_key] = bk_value
-
-                if bk_value > max_bookmarks[stream][stream_bookmark_key]:
-                    max_bookmarks[stream][stream_bookmark_key] = bk_value
+            # `transactions` is child stream of `orders` stream which is incremental.
+            # We are writing a separate bookmark for the child stream in which we are storing 
+            # the bookmark based on the parent's replication key.
+            # But, we are not using any fields from the child record for it.
+            # That's why the `transactions` stream does not have replication_key but still it is incremental.
+            if stream not in ('transactions'):
+                upsert_messages = [m for m in batch.get('messages') if m['action'] == 'upsert']
+                stream_bookmark_key = self.expected_replication_keys().get(stream, set())
+                assert len(stream_bookmark_key) == 1  # There shouldn't be a compound replication key
```

Comment:
On line #257 and #285, tuple ('transactions') is hard coded. instead can we declare a variable to initialise this tuple? Advantage being in future if we get similar child stream, we can simply add it in the tuple without making any more changes.

Reply:
Yes right, created a variable and used that

Reply:
Please update the same in test files as well.

- tests/test_bookmarks.py
- tests/test_bookmarks_updated.py
- tests/test_discovery.py
- tests/test_start_date.py

Reply:
Updated in all th files